### DISCUSSION TITLE
Parse Error unexpected 'random letter' , expecting button

### DISCUSSION THREAD

**Discussion by ScyMarshall** [kmonad/kmonad#627] [Q&A] [answered]
Edit 2:
Hi guys,
I believe that I have got the kink out of it.
It works now, it compiles.
I just had to change nearly every keystroke.......
Now it makes stress when I am pushing the button for "s" on the home row.Former versions made a tab, this version jumps out of the layer.
As source, I chose qwerty, but actually, I write on a german qwertz board.
I will try to change that as well in hopes that it works,then.
---------------------------------
Original Message below
---------------------------------
Hi guys,

I am trying to build a KOY style keyboard layout, with kind Thank you to https://github.com/HenryJobst
After many mistakes, I decided to start from scratch, letter for letter.
I need to use many shortcuts to get around as a german.

Well, now if I want to start kmonad, it prints this message:

`kmonad.exe: Parse error at 144:5:
       | 
144 |     di  #(ralt d i)  ;;diameter ()
       |     ^
unexpected 'd'
expecting button
`

Even if I comment out the whole line, it finds some other line to whine about.

Any tips or ideas?
This is the file...
https://github.com/ScyMarshall/Kmonad-with-KOY-layout/blob/04a959d39d2e57b9c3b3c2cdbd8bea8c00528196/liebeskindV2.kbd
If anything is unclear, I will try to explain this abomination^^
Thank you, guys!

EDIT 1:
It is a whole block that sucks.
I commented it completly out, now it is this message:
`kmonad.exe: Parse error at 159:2
159 | ) 
           ^
unrecognized compose char`

**Comment by gerhard-h** [comment, marked as answer]
does triggering wincompose from kmonad even work for you **?**
on my PC `#(compose-key d i) `  is bypassing wincompose and just outputs `di` 

I suppose it is easier to run the **standard** US keyboard layout  inside Windows to get going with kmonad and also because it reduces compatibility issues.

I myself kept using DE keyboard layout inside windows because I didn't want to care about shifted keys like S-3 being # instead of §  and so on. https://github.com/gerhard-h/kmonad-keymaps/blob/master/qwertz.kbd

the tricky part to start of using DE Keymap is to shuffle around special chars in defsrc like so:
```

(defsrc
  esc  F1   F2   F3   F4   F5   F6   F7   F8   F9   F10  F11  F12        ssrq slck pause
  \    1    2    3    4    5    6    7    8    9    0    [    ]    bspc  ins  home pgup  nlck kp/  kp*  kp-
  tab  q    w    e    r    t    z    u    i    o    p    ;    =    ret   del  end  pgdn  kp7  kp8  kp9  kp+
  caps a    s    d    f    g    h    j    k    l    grv  '    /                          kp4  kp5  kp6
  lsft 102d y    x    c    v    b    n    m    ,    .    -    rsft            up         kp1  kp2  kp3  kprt
  lctl lmet lalt           spc                      ralt cmp  rctl       left down rght  kp0  kp.
)
```

to resemble this 

```

(defsrc
  esc  F1   F2   F3   F4   F5   F6   F7   F8   F9   F10  F11  F12        ssrq slck pause
  ^    1    2    3    4    5    6    7    8    9    0    ß    ´    bspc  ins  home pgup  nlck kp/  kp*  kp-
  tab  q    w    e    r    t    z    u    i    o    p    ü    +    ret   del  end  pgdn  kp7  kp8  kp9  kp+
  caps a    s    d    f    g    h    j    k    l    ö    ä     #                         kp4  kp5  kp6
  lsft 102d y    x    c    v    b    n    m    ,    .    -    rsft            up         kp1  kp2  kp3  kprt
  lctl lmet lalt           spc                      ralt cmp  rctl       left down rght  kp0  kp.
)
```
(I couldn't find any logic why e.g. `;` must be replaced by `grv` to get `ö `)
